rest/client/rest.go

```diff
@@ -20,6 +20,27 @@ import (
 	"github.com/pkg/errors"
 )
 
+// respErrorf attempts to read a gimlet.ErrorResponse from the response body
+// JSON. If successful, it returns the gimlet.ErrorResponse wrapped with the
+// HTTP status code and the formatted error message. Otherwise, it returns an
+// error message with the HTTP status and raw response body.
+func respErrorf(resp *http.Response, format string, args ...interface{}) error {
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		err = errors.Wrapf(err, "status code %d", resp.StatusCode)
+		return errors.Wrap(err, "could not read response body")
+	}
+
+	respErr := gimlet.ErrorResponse{}
+	if err = json.Unmarshal(b, &respErr); err != nil {
```

Comment:
I could be wrong, but I believe this logic doesn't address the scenario you described because that will return json with fields not in this struct, and this will silently ignore extra fields (so that this branch is only hit if the response is not actually json)

Reply:
Words are hard - this is supposed to handle the case where a response body is _not_ JSON. It doesn't deal with the question of it being JSON but with extra fields.